src/TraceEvent/TraceLog.cs

```diff
@@ -154,10 +154,192 @@ public static TraceLog OpenOrConvert(string etlOrEtlxFilePath, TraceLogOptions o
+        /// </summary>
+        private void SetupInitialRundownCallbacks(EventPipeEventSource rawEvents)
+        {
+            rawEvents.HeadersDeserialized += delegate () { CopyHeadersFrom(rawEvents); };
+
+            var clrRundownParser = new ClrRundownTraceEventParser(rawEvents);
+            Action<ModuleLoadUnloadTraceData> onLoaderRundown = delegate (ModuleLoadUnloadTraceData data)
+            {
+                TraceProcess process = processes.GetOrCreateProcess(data.ProcessID, data.TimeStampQPC);
+                process.LoadedModules.ManagedModuleLoadOrUnload(data, false, true);
+            };
+
+            clrRundownParser.LoaderModuleDCStop += onLoaderRundown;
+            clrRundownParser.LoaderModuleDCStart += onLoaderRundown;
+
+            clrRundownParser.MethodILToNativeMapDCStop += delegate (MethodILToNativeMapTraceData data)
+            {
+                codeAddresses.AddILMapping(data);
+            };
+
+            clrRundownParser.MethodDCStopVerbose += delegate (MethodLoadUnloadVerboseTraceData data)
+            {
+                // Note: we need this also for non-jitted methods, otherwise we won't resolve some frames, for example:
+                //      "System.Private.CoreLib.il" - "System.Threading.Tasks.Task.Wait()"
+                TraceProcess process = processes.GetOrCreateProcess(data.ProcessID, data.TimeStampQPC);
+                process.InsertJITTEDMethod(data.MethodStartAddress, data.MethodSize, delegate ()
+                {
+                    TraceManagedModule module = process.LoadedModules.GetOrCreateManagedModule(data.ModuleID, data.TimeStampQPC);
+                    MethodIndex methodIndex = CodeAddresses.Methods.NewMethod(GetFullName(data), module.ModuleFile.ModuleFileIndex, data.MethodToken);
+                    return new TraceProcess.MethodLookupInfo(data.MethodStartAddress, data.MethodSize, methodIndex);
+                });
+            };
+        }
+
+        private unsafe void OnAllEventPipeEventsRealTime(TraceEvent data)
+        {
+            TraceEventCounts countForEvent = Stats.GetEventCounts(data);
+            // Debug.Assert((int)data.EventIndex == eventCount);
+            countForEvent.m_count++;
+            countForEvent.m_eventDataLenTotal += data.EventDataLength;
+
+            // Remember past events so we can hook up stacks to them.
+            data.eventIndex = (EventIndex)eventCount;
+            pastEventInfo.LogEvent(data, data.eventIndex, countForEvent);
+
+            // currentID is used by the dispatcher to define the EventIndex.  Make sure at both sources have the
+            // same notion of what that is if we have two dispatcher.
+            if (rawEventSourceToConvert != null)
+            {
+                rawEventSourceToConvert.currentID = (EventIndex)eventCount;
+            }
+
+            // Skip samples from the idle thread.
+            if (data.ProcessID == 0 && data is SampledProfileTraceData)
```

Comment:
This should not happen because we'll always be attached to a real process, and there is no idle process for event pipe.  You can remove this check.

Reply:
right, thanks for the insight